openedx_webhooks/tasks/github.py

```diff
@@ -0,0 +1,336 @@
+# coding=utf-8
+from __future__ import unicode_literals, print_function
+
+import json
+import re
+from datetime import date
+
+from iso8601 import parse_date
+from flask import render_template
+
+from openedx_webhooks import sentry, celery
+from openedx_webhooks.tasks import logger, github, jira
+from openedx_webhooks.info import (
+    get_people_file, is_internal_pull_request, is_contractor_pull_request
+)
+from openedx_webhooks.utils import memoize, paginated_get
+from openedx_webhooks.jira_views import get_jira_custom_fields
+
+
+@celery.task
+def pull_request_opened(pull_request, ignore_internal=True, check_contractor=True):
+    pr = pull_request
+    user = pr["user"]["login"].decode('utf-8')
+    repo = pr["base"]["repo"]["full_name"]
+    num = pr["number"]
+    if ignore_internal and is_internal_pull_request(pr, session=github):
+        # not an open source pull request, don't create an issue for it
+        logger.info(
+            "@{user} opened PR #{num} against {repo} (internal PR)".format(
+                user=user, repo=repo, num=num,
+            ),
+        )
+        return "internal pull request"
+
+    if check_contractor and is_contractor_pull_request(pr, session=github):
+        # have we already left a contractor comment?
+        if has_contractor_comment(pr, session=github):
+            return "contract pull request (commented)"
+
+        # don't create a JIRA issue, but leave a comment
+        comment = {
+            "body": github_contractor_pr_comment(pr),
+        }
+        url = "/repos/{repo}/issues/{num}/comments".format(
+            repo=repo, num=num,
+        )
+        comment_resp = github.post(url, json=comment)
+        comment_resp.raise_for_status()
+        return "contractor pull request"
+
+    issue_key = get_jira_issue_key(pr)
+    if issue_key:
+        msg = "Already created {key} for PR #{num} against {repo}".format(
+            key=issue_key,
+            num=pr["number"],
+            repo=pr["base"]["repo"]["full_name"],
+        )
+        logger.info(msg)
+        return msg
+
+    repo = pr["base"]["repo"]["full_name"].decode('utf-8')
+    people = get_people_file(session=github)
+    custom_fields = get_jira_custom_fields(jira)
+
+    if user in people:
+        user_name = people[user].get("name", "")
+    else:
```

Comment:
Perhaps better as:
```python
user_name = people[user].get("name", "")
if not user_name:
    user_resp = github.get(pr["user"]["url"])
    ...
```

this way we don't end up with a blank user_name ever, at worst we get their github username.